### DISCUSSION TITLE
Add support for a `insert_final_newline` setting

### DISCUSSION THREAD

**Discussion by aslilac** [helix-editor/helix#8008] [Ideas] [answered]
...which would do basically what it sounds like. If the file does not end with a newline, then add one before actually writing to the disk when invoking `:w`.

There's lot of precedent for a setting like this such as `"files.insertFinalNewline` in VSCode or the more general `insert_final_newline` setting specified by EditorConfig.

I'm very used to having VSCode do this for me, and I very much miss it when using Helix. I very often forget to add it when making a new file, only to sob quietly when I see the red little "you dun goofed" icon on Github when looking at it later.

**Comment by zqianem** [comment, marked as answer]
Done now that https://github.com/helix-editor/helix/pull/8157 is merged 🎉

**Reply by aslilac** [reply]
oh sweet!
